### PR TITLE
Hourglass Quest

### PR DESCRIPTION
**Pull request by SabreCat**
[//]: # (Note: See http://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

* Introduces the ability to buy Quests marked with a category of `timeTravelers` from the Time Travelers Shop using Mystic Hourglasses.
* Adds the first such quest, "Mysterious Mechanical Marvels", for Robot pet eggs

### CODE REVIEW

File: website/server/controllers/api-v3/user.js [hunk 1]
```diff
@@ -1000,6 +1000,10 @@ api.userPurchaseHourglass = {
   url: '/user/purchase-hourglass/:type/:key',
   async handler (req, res) {
     let user = res.locals.user;
+    req.params.hourglass = true;
```
Comment (paglias) [inline website/server/controllers/api-v3/user.js]: Hmm this feels hacky... `req.body.quantity` is lacking validation on it being a number and apidocs and I'd not change the original `req.params` object

Is it necessary for quantity to be in req.params? Cannot you access `req.body.quantity` in `common.ops.buy`?

File: website/client/store/actions/shops.js [hunk 2]
```diff
@@ -112,12 +112,13 @@ export function purchaseMysterySet (store, params) {
...
 export function purchaseHourglassItem (store, params) {
+  const quantity = params.quantity || 1;
   const user = store.state.user.data;
-  let opResult = hourglassPurchaseOp(user, {params});
+  let opResult = hourglassPurchaseOp(user, {params, quantity});
 
   return {
     result: opResult,
-    httpCall: axios.post(`/api/v4/user/purchase-hourglass/${params.type}/${params.key}`),
+    httpCall: axios.post(`/api/v4/user/purchase-hourglass/${params.type}/${params.key}`, {quantity}),
```
Comment (negue) [inline website/client/store/actions/shops.js:121]: if quantity would be also a `/{params.quantity}`-param, then it wouldn't have to be in body

Comment (SabreCat) [reply website/client/store/actions/shops.js:121]: That would require creating a whole new route, wouldn't it?

File: website/server/controllers/api-v3/user.js (hunk 1 above)
Comment (SabreCat) [reply website/server/controllers/api-v3/user.js]: Hah, this was modeled after some other routes where we do weird things with `req.params.quantity`, but you're absolutely right this was one of the things I was nervous about. 78de2a5 better?

File: website/client/store/actions/shops.js (hunk 2 above)
Comment (negue) [reply website/client/store/actions/shops.js:121]: I think when you add a param it would still accept the route (without the newly added one)

**Pull request merged**